### PR TITLE
HCX-119:HCX Participant Code and End Point URL validations

### PR DESCRIPTION
**Pull request by tarento-dharmateja**

### CODE REVIEW

File: api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java [hunk 1]
```diff
@@ -92,9 +99,19 @@ protected void validateCondition(Boolean condition, ErrorCodes errorcode, String
     }
 
     protected void validateParticipant(Map<String,Object> details, ErrorCodes code, String participant) throws ClientException {
+        String participantCode = "1" + (String) details.get("participant_code");
```
Comment (maheshkumargangula) [inline api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java:102]: why are we appending the code with "1"?

File: api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java [hunk 2]
```diff
@@ -92,9 +99,19 @@ protected void validateCondition(Boolean condition, ErrorCodes errorcode, String
...
+        ArrayList<String> roles = (ArrayList) details.get("roles");
+
         if(details.isEmpty()){
             throw new ClientException(code, participant + " does not exist in registry");
-        } else if(StringUtils.equals((String) details.get("status"), BLOCKED) || StringUtils.equals((String) details.get("status"), INACTIVE)){
+        }
+        if(participantCode.equals(hcxCode)) {
+            throw new ClientException(code, participant + " should not be sent as recipient in the incoming requests");
+        }
+        if(roles.contains(hcxRoles)) {
```
Comment (maheshkumargangula) [inline api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java:111]: hcxRoles is a single value or multiple. Are we handling the case sensitive input?

File: hcx-apis/src/main/resources/application.yml [hunk 3]
```diff
@@ -56,3 +56,8 @@ redirect:
   headers:
     mandatory: ${redirect_mandatory_headers:x-hcx-sender_code, x-hcx-recipient_code, x-hcx-api_call_id, x-hcx-timestamp, x-hcx-correlation_id, x-hcx-status, x-hcx-redirect_to}
     optional: ${redirect_optional_headers:x-hcx-workflow_id}
+
+#hcx not allowed urls
+hcx:
+  urls:
+    notallowed: ${not_allowed_urls:http://a54c5bc648f1a41b8871b77ac01060ed-1840123973.ap-south-1.elb.amazonaws.com:8080,http://a851b026063934683964f22c6732f177-1969328968.ap-south-1.elb.amazonaws.com:8095}
```
Comment (maheshkumargangula) [inline hcx-apis/src/main/resources/application.yml:63]: This should be the human-readable public URL.

**Review by maheshkumargangula — CHANGES_REQUESTED** [review]

File: api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java (hunk 2 above)
Comment (tarento-dharmateja) [reply api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java:111]: single value

File: api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java (hunk 1 above)
Comment (tarento-dharmateja) [reply api-gateway/src/main/java/org/swasth/apigateway/models/BaseRequest.java:102]: in registry response we get participant code without "1".

File: hcx-apis/src/main/resources/application.yml (hunk 3 above)
Comment (tarento-dharmateja) [reply hcx-apis/src/main/resources/application.yml:63]: okay i'll change

**Comment by maheshkumargangula** [comment]
@tarento-dharmateja - Please improve the code coverage. It is very less. Also, still the case sensitivity not taken care for roles.

**Review by maheshkumargangula — APPROVED** [review]

**Pull request merged**